android-core/src/main/java/com/mparticle/internal/database/services/MParticleDBManager.java

```diff
@@ -311,12 +275,6 @@ private void createUploads(Map<BatchId, MessageBatch> uploadMessagesByBatchId, M
                 }
 
                 UploadService.insertUpload(db, batch, configManager.getApiKey());
-                //if this was to process session history, or
-                //if we're never going to process history AND
-                //this batch contains a previous session, then delete the session.
-                if (!historyMessages && !sessionHistoryEnabled) {
-                    cleanSessions(currentSessionId);
```

Comment:
we need line no 318 cleanSessions(currentSessionId);

Reply:
done